src/components/SearchPage.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { useEffect } from "react";
+import { Box, Container, Typography } from "@material-ui/core";
+import { useSearch } from "../api/WordsAPI";
+import useRouteQuery from "../hooks/useRouteQuery";
+import _ from "lodash";
+import SearchResult, { SearchItem } from "./SearchResult";
+import SearchBar from "./SearchBar";
+import { useHistory } from "react-router-dom";
+
+const NUMBER_OF_RESULT = 50;
+
+const SearchPage: React.FC = () => {
+  const routeQuery = useRouteQuery();
+  const wordLabel = routeQuery.get("label");
+  const {
+    data = [],
+    isSuccess,
+    isLoading,
+    isError,
+  } = useSearch(wordLabel ?? undefined);
+  const history = useHistory();
+
+  useEffect(() => {
```

Comment:
If I'm reading this correctly, this hook is called every time the component is rerendered. At the very least it should depend on the `isSuccess` variable.

Suggestion for refactoring after the MVP is done: the principle of post-checking whether the query is a word or term and replacing history is not ideal, it may lead to unwanted recomputing of the whole component tree from the router switch. It would be better if the search box directed users to the right URLs instead so that there are no extra steps.